apps/slackbot/bolt_app.py

```diff
@@ -123,21 +127,43 @@ def get_response(question, previous_messages):
 def event_test(client, say, event):
     question = event["text"]
     thread_ts = event.get("thread_ts", None) or event["ts"]
-    replies = client.conversations_replies(channel=event["channel"], ts=thread_ts)
+    replies = client.conversations_replies(
+        channel=event["channel"], ts=thread_ts)
     previous_messages = replies["messages"][:-1]
 
-    # used to reconstruct the question. if the question contains a link recreate
-    # them so that they contain scraped and summarized content of the link
-    reconstructor = PromptReconstructor(
-        question=question, slack_message=[replies["messages"][-1]]
-    )
-    question = reconstructor.reconstruct_prompt()
-    results, verbose_message = get_response(question, previous_messages)
-   
-    say(results, thread_ts=thread_ts)
+    input_message = replies['messages'][-1]
+    user_id = input_message['user']
+    team_id = input_message['team']
+    combined_id = user_id+"_"+team_id
+
+    user_db = UserUsageTracker(max_daily_token=20000)
+
+    usage_cheker = user_db.check_usage(
+        user_id=user_id, combined_id=combined_id, token_ammount=count_string_tokens(question, "gpt-3.5-turbo"))
+    can_excute = usage_cheker['can_excute']
+
+    user_db.close_connection()
+    # only will be excuted if the user don't pass the daily limit
+    # the daily limit is calculated based on the user's usage in a workspace 
+    # users with a daily limitation can be allowed to use in a different workspace
+    if can_excute:
+        # used to reconstruct the question. if the question contains a link recreate
+        # them so that they contain scraped and summerized content of the link
+        reconstructor = PromptReconstructor(
+            question=question, slack_message=[replies["messages"][-1]]
+        )
+        question = reconstructor.reconstruct_prompt(
+            user_id=user_id, team_id=team_id)
 
-    if contains_verbose(question):
-        say(f"#verbose message: \n```{verbose_message}```", thread_ts=thread_ts)
+        results, verbose_message = get_response(question=question, previous_messages=previous_messages,
+                                                team_id=team_id, user_id=user_id)
+        say(results, thread_ts=thread_ts)
+
+        if contains_verbose(question):
+            say(f"#verbose message: \n```{verbose_message}```",
+                thread_ts=thread_ts)
+    else:
+        say("I apologize for the inconvenience, but it seems that you have exceeded your daily token limit. As a result, you will need to try again after 24 hours. Thank you for your understanding.",thread_ts=thread_ts)
```

Comment:
We should extract this as a variable or something so it is configurable. Maybe in config.py?